lib/tasks/journeys_report.rake

```diff
@@ -2,60 +2,98 @@ require 'csv'
 require 'json'
 require 'aws-sdk-s3'
 
+def get_last_run(bucket_name)
+  s3 = Aws::S3::Resource.new
+  if s3.bucket(bucket_name).object('bookmark').exists?
+    bookmark = s3.bucket(bucket_name).object('bookmark').get
+    last_run = JSON.parse(bookmark.body.read)['last_run']
+    last_run_date = DateTime.parse(last_run)
+  else
+    last_run_date = DateTime.strptime('0', '%s')
+  end
+  last_run_date
+end
+
+def update_bookmark(bucket_name)
+  s3 = Aws::S3::Resource.new
+  bookmark = { 'last_run': DateTime.now.utc.to_s }
+  s3.bucket(bucket_name).put_object(key: 'bookmark', body: JSON[bookmark])
+end
+
 namespace :journeys do
   desc 'Exports a JSON report of all journeys'
   task json_report: :environment do
     timestamp = DateTime.now.utc.to_i
+    bucket_name = ENV['S3_REPORTING_BUCKET_NAME'] || raise('S3_REPORTING_BUCKET_NAME not defined')
+
+    if ENV['RAILS_ENV'].to_s == 'development' || ENV['RAILS_ENV'].to_s == ''
+      Aws.config.update(
+        endpoint: 'http://localhost:4572',
+        credentials: Aws::Credentials.new('fakeid', 'fakesecret'),
+        region: 'eu-west-2',
+        force_path_style: true,
+      )
+    end
+
+    last_report_date = get_last_run(bucket_name)
```

Comment:
consider the logic for picking moves between certain times, and also the time it takes for the exporter to run. You'll need to ensure there are no duplicates and no missed moves between runs.

Reply:
What happens if moves are created whilst the process is running ? I think it will be necessary to define a clear from_datetime and to_datetime and only pick moves within that range

Reply:
Actually, thinking about this further; its possible for a move created last week to have a new journey added today: if we just filter on move.created_at time, we will miss the new journey. So, filtering needs more thought!

Reply:
Yes, agreed.  I'll change to `created_at` for now, just to get the logic roughly working and we'll need to revisit to catch those edge cases.

Reply:
Using the effecting report time with the new approach.

Reply:
> if we just filter on move.created_at time,

Would updated_at be a more suitable choice?

Reply:
so, unfortunately, updating a journey attached to a move does not update the parent move record.

However, its pretty trivial to add this functionality: in the journey model (`app/models/journey`), change `belongs_to :move` to `belongs_to :move, touch: true`

Now, whenever a journey is updated, so will its parent move. Rails magic!

Reply:
Will do that in a separate PR, as it affects the main app as well.